### PR TITLE
Fix: grep(~) will change the original result when ignores case

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Insensitive grep(~) will change the original result when ignores the case.

run:
```
rizin bins/pe/libzmq-v100-mt-4_0_4.dll
fl~+0x1004e0f4[2]
```
expected: 
```
sym.imp.MSVCP100.dll_public:_void___thiscall_std::basic_ostream_char__struct_std::char_traits_char__::_Osfx_void
```

actual:
```
sym.imp.msvcp100.dll_public:_void___thiscall_std::basic_ostream_char__struct_std::char_traits_char__::_osfx_void
```

**Test plan**

CI is green.

**Closing issues**

...

### CODE REVIEW

**Review by ret2libc — COMMENTED** [review]
Please add a small unit test for `rz_str_case_nstr` in test_str.c

File: librz/util/str.c [hunk 1]
```diff
@@ -860,6 +860,26 @@ RZ_API const char *rz_str_nstr(const char *s, const char *find, int slen) {
 	return (char *)s;
 }
 
+/**
+ * \brief Finds the first occurrence of \p find in \p s, ignore case.
+ * \param s pointer to the string to examine
+ * \param find pointer to the string to search for
+ * \param slen the maximum number of characters to search
+ */
+RZ_API const char *rz_str_case_nstr(const char *s, const char *find, int slen) {
```
Comment (XVilka) [inline librz/util/str.c]: check for NULL with asserts, add `RZ_NONNULL`

File: librz/util/str.c [hunk 2]
```diff
@@ -2739,18 +2759,32 @@ RZ_API size_t rz_str_len_utf8_ansi(const char *str) {
...
 // XXX must find across the ansi tags, as well as support utf8
-RZ_API const char *rz_strstr_ansi(const char *a, const char *b) {
+/**
+ * \brief Finds the first occurrence of \p b in \p a, skip ansi sequence.
+ * \param a pointer to the string to examine
+ * \param b pointer to the string to search for
+ * \param icase whether ignore case
+ */
+RZ_API const char *rz_strstr_ansi(const char *a, const char *b, bool icase) {
 	const char *ch, *p = a;
```
Comment (XVilka) [inline librz/util/str.c:2771]: Same

**Review by XVilka — CHANGES_REQUESTED** [review]

**Review by kazarmy — CHANGES_REQUESTED** [review]
https://github.com/rizinorg/rizin/blob/13cd3942d12b61911e27ab82baef045adf57d77c/test/db/formats/demangling_bin#L71-L73

Modify the test as well.

**Comment by PeiweiHu** [comment]
Thanks for the constructive comments. Apply all suggestions.

**Review by XVilka — APPROVED** [review]

**Review by kazarmy — APPROVED** [review]

**Pull request merged**